.github/workflows/build_and_test.yml

```diff
@@ -9,13 +9,16 @@ env:
   BUILD_DIR: Dist
   GCOVR_FLAGS: --gcov-ignore-parse-errors --exclude-throw-branches --filter Common --filter Pcap --filter Packet --xml
 
+permissions:
+  contents: read
+
 jobs:
   pre-commit:
     runs-on: ubuntu-latest
     container: seladb/alpine317
     steps:
       - name: Checkout code
-        uses: actions/checkout@v3
+        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
```

Comment:
I don't really like the fact that they change the versions to a hash everywhere 😕

Reply:
Luckily dependabot updates it automatically with the updated hash. So, there is no need manual replacement/check.

Reply:
Agree, we already trust this CI actions, if they want to mess up they already can do it.
Having a Hash instead of a tag will not make a hug difference...

Reply:
@egecetin I'm not sure I understand. Dependabot only uses versions and not hashes, no?

I'm not sure how the StepSecurity tool works, but it uses hash instead of versions

Reply:
If dependabot see version tags are used then it updates as version tags otherwise if it see hashes are used it updates the hashes with the version comment like this PR. So if we merge this dependabot will also use hashes instead of version tags.

Reply:
got it, I didn't know that 👍